pkg/client/elasticsearch/query.go

```diff
@@ -291,30 +289,34 @@ func getMetricDocument(
 	metricSelector labels.Selector,
 	doc map[string]interface{},
 ) (map[string]interface{}, error) {
-	metaHits, hasMetaHits := doc["hits"]
-	if !hasMetaHits {
-		klog.Errorf("no hits in %v", metaHits)
+	metaHits, ok := doc["hits"]
+	if !ok {
 		return nil, provider.NewMetricNotFoundForSelectorError(info.GroupResource, info.Metric, name.Name, metricSelector)
 	}
 
-	var documents interface{}
-	if hits, ok := metaHits.(map[string]interface{}); ok {
-		documents, _ = hits["hits"]
-	} else {
+	hits, ok := metaHits.(map[string]interface{})
```

Comment:
These changes lgtm but I'm a bit concerned as there are not covered by any unit test.

Reply:
Unit test added in aa3cff1.